### PR TITLE
All motion params

### LINKED ISSUE #1576: ✨ Condense motion numbers into one TSV
### Related problem

Currently, CPAC outputs a range of motion-derived measures across several disparate text/1D files:

```
sub-*_ses-*_task-rest_run-*_dvars.1D
sub-*_ses-*_task-rest_run-*_framewise-displacement-jenkinson.1D
sub-*_ses-*_task-rest_run-*_framewise-displacement-power.1D
sub-*_ses-*_task-rest_run-*_max-displacement.1D
sub-*_ses-*_task-rest_run-*_rels-displacement.1D
sub-*_ses-*_task-rest_run-*_motion-params.txt
sub-*_ses-*_task-rest_run-*_movement-parameters.1D
sub-*_ses-*_task-rest_run-*_power-params.txt
```
(The `rels-displacement.1D` output is only produced when running FSL MCFLIRT for motion correction).

The problems:

- This is confusing, users cannot easily tell which files will contain what information. All of the names are vaguely similar.
- Adds to the large amount of files in the `func` output directory.
- Somewhat wasteful as some of the files contain a singular value.
- They should be `.tsv` files to comply with [BIDS-Derivatives](https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#tabular-files).

### Proposed feature

It would be good to condense all of this data into 3 files. There are three kinds:

1. One row of numbers with a header denoting each one. We can merge these into one file and change it to a `.tsv`:
```
sub-*_ses-*_task-rest_run-*_motion-params.txt
sub-*_ses-*_task-rest_run-*_power-params.txt
```

2. A 1D file with a single vector, with a row/value per time point from the time series. Again, merge these files, but also create a new header describing each vector, and make it a `.tsv` file:
```
sub-*_ses-*_task-rest_run-*_dvars.1D
sub-*_ses-*_task-rest_run-*_framewise-displacement-jenkinson.1D
sub-*_ses-*_task-rest_run-*_framewise-displacement-power.1D
sub-*_ses-*_task-rest_run-*_max-displacement.1D
sub-*_ses-*_task-rest_run-*_rels-displacement.1D
```

3. The `movement-parameters.1D` output, which is often a 6xN matrix, with N = timepoints. This file is used widely downstream as its own output, in the format it is written in, so this can remain the same, although we can also convert it to a `.tsv` file:
```
sub-*_ses-*_task-rest_run-*_movement-parameters.1D
```

There is **one really important caveat**- all of this data should still remain in the resource pool internally; we should only [remove them as written outputs](https://github.com/FCP-INDI/C-PAC/blob/07d59f63ad716485ab0cf96e84f5d9a531312e9e/CPAC/resources/cpac_outputs.tsv#L95-L104). Within the node blocks, many blocks need to [easily access singular motion measures](https://github.com/FCP-INDI/C-PAC/blob/07d59f63ad716485ab0cf96e84f5d9a531312e9e/CPAC/nuisance/nuisance.py#L2272-L2273).

So this would involve simply _adding_ a new resource pool resource for each of the two new files.

An alternative would be to set up a system where you call in one of the TSV data files and then be able to request which column, but this would start to require contributors to know about this when designing a sub-part of the system; would prefer to avoid this.

**Naming of the two new files:** open to suggestions.

### Acceptance criteria

- [ ] Three motion files are produced when running motion correction.
- [ ] Check both `AFNI 3dvolreg` and `FSL MCFLIRT` runs.
- [ ] The other files no longer get written to the output directory.

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tergeorge**
## Fixes

Fixes #1576 by @sgiavasis 

## Description

The pull request is used to condense the motion parameters into a two .tsv files.
## Technical details

changes made to func_preproc.py and generate_motion_statistics.py. 
All .1D files mentioned below combined into 1 .tsv file:
sub-*_ses-*_task-rest_run-*_framewise-displacement-jenkinson.1D
sub-*_ses-*_task-rest_run-*_framewise-displacement-power.1D
sub-*_ses-*_task-rest_run-*_max-displacement.1D
sub-*_ses-*_task-rest_run-*_rels-displacement.1D
sub-*_ses-*_task-rest_run-*_movement-parameters.1D

All 2D matrix values mentioned below combined into 2nd .tsv file:
sub-*_ses-*_task-rest_run-*_motion-params.txt
sub-*_ses-*_task-rest_run-*_power-params.txt

The new files generated are motion.tsv and summary-motion.tsv. 

## Tests
Run the PR with any BIDS dataset and the two .tsv files should be generated in the working and output directory.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository. 
- [x] My commit messages follow best practices.
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I updated the changelog.
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sgiavasis** [comment]
There seem to be four files that might have been accidentally created - viewing the Files Changed will show 4 empty files, with names of our usual files but in the wrong places:
- [CPAC/func_preproc.py](https://github.com/FCP-INDI/C-PAC/pull/1624/files#diff-ba547dbede7c3c2b0773cce571d17e5e09df3f7faf4a88082bb255632537397b)
- [CPAC/generate_motion_statistics.py](https://github.com/FCP-INDI/C-PAC/pull/1624/files#diff-51bd87f595b264e78f2a3b6836a6391686998c8a1f3448fa88592f3307dfcc84)
- [func_preproc.py](https://github.com/FCP-INDI/C-PAC/pull/1624/files#diff-6d2f060650125f1ac61ae79ae8bd985e81ef8e7d2d4dcf72d524517e5d3e57f3)
- [generate_motion_statistics.py](https://github.com/FCP-INDI/C-PAC/pull/1624/files#diff-99e379275aa2a7dc5945f1b6edc50785a383f76a9fb0b77a44eedd63949c0d7c)

You'll need to remove these before merging.

**Comment by tergeorge** [comment]
Deleted the empty files and tested and updated the PR.

**Review by shnizzedy — CHANGES_REQUESTED** [review]

File: CPAC/resources/cpac_outputs.tsv
Comment (tergeorge) [reply CPAC/resources/cpac_outputs.tsv:193]: Not sure how it ended up deleting those!!

Comment (tergeorge) [reply CPAC/resources/cpac_outputs.tsv:193]: we need to delete these though framewise-displacement-jenkinson, framewise-displacement-power, max-displacement, motion-params, power-params and rels-displacement

Comment (shnizzedy) [reply CPAC/resources/cpac_outputs.tsv]: And `dvars`?

**Review by shnizzedy — CHANGES_REQUESTED** [review]
Alright, I finally finished writing https://github.com/FCP-INDI/C-PAC/pull/1755; this one should be easier to review after that one.

File: CPAC/resources/cpac_outputs.tsv
Comment (shnizzedy) [reply CPAC/resources/cpac_outputs.tsv]: Resolved in https://github.com/FCP-INDI/C-PAC/pull/1755

File: CPAC/generate_motion_statistics/generate_motion_statistics.py
Comment (tergeorge) [reply CPAC/generate_motion_statistics/generate_motion_statistics.py]: oh yes I forgot to delete those path assignment lines from the code. Thanks for looking into it.

File: CPAC/generate_motion_statistics/generate_motion_statistics.py
Comment (shnizzedy) [reply CPAC/generate_motion_statistics/generate_motion_statistics.py]: Resolved in ae7861d22b3948829dc2c74936c04a9982a9578e

File: CPAC/generate_motion_statistics/generate_motion_statistics.py
Comment (shnizzedy) [reply CPAC/generate_motion_statistics/generate_motion_statistics.py]: Resolved in ae7861d22b3948829dc2c74936c04a9982a9578e

File: CPAC/generate_motion_statistics/generate_motion_statistics.py
Comment (shnizzedy) [inline CPAC/generate_motion_statistics/generate_motion_statistics.py:461]: I think you still need https://github.com/FCP-INDI/C-PAC/blob/2cbd2d340efec3ee0aa5179e5f26067d1c68dc4e/CPAC/generate_motion_statistics/generate_motion_statistics.py#L488

**Review by shnizzedy — CHANGES_REQUESTED** [review]

File: CPAC/generate_motion_statistics/generate_motion_statistics.py
Comment (tergeorge) [reply CPAC/generate_motion_statistics/generate_motion_statistics.py:461]: oh right! I think I deleted it by mistake. Thanks for looking through it.

**Review by shnizzedy — APPROVED** [review]

**Review by shnizzedy — CHANGES_REQUESTED** [review]
I think this is most of the way there. Remaining issues that I see:

- [ ] MCFLIRT crash needs to be addressed
- [ ] Old one-off outputs need to be removed from `cpac_outputs.tsv`
- [ ] I think we want `summary-motion` to be `desc-summary_motion` for BIDS reasons

File: CPAC/generate_motion_statistics/generate_motion_statistics.py
Comment (tergeorge) [reply CPAC/generate_motion_statistics/generate_motion_statistics.py:481]: Added `rels_displacement` as a resource to `calc_motion_parameters `.

File: run-with-freesurfer.sh
Comment (tergeorge) [reply run-with-freesurfer.sh:1]: Deleted the empty file.

File: CPAC/resources/cpac_outputs.tsv
Comment (tergeorge) [reply CPAC/resources/cpac_outputs.tsv:1]: Deleted the loose files from `cpac_outputs.tsv`.  I kept `motion-filter-plot` from the list.

File: CPAC/resources/cpac_outputs.tsv
Comment (tergeorge) [reply CPAC/resources/cpac_outputs.tsv]: changed summary-motion to desc-summary_motion.

File: CPAC/func_preproc/func_preproc.py
Comment (tergeorge) [reply CPAC/func_preproc/func_preproc.py:1385]: I am not sure about this.

File: CPAC/resources/cpac_outputs.tsv
Comment (shnizzedy) [reply CPAC/resources/cpac_outputs.tsv:214]: Yeah that row was missing 2 columns

**Comment by shnizzedy** [comment]
> This is how the output files are named `sub-0050952_ses-1_task-rest_run-1_desc-motion_motion.tsv` and `sub-0050952_ses-1_task-rest_run-1_desc-descSummaryMotion_motion.tsv` . It looks like `desc` gets added to the output file name.

― https://github.com/FCP-INDI/C-PAC/pull/1624#issuecomment-1610228675


Yep, they get added in https://github.com/FCP-INDI/C-PAC/blob/35d669bd472006898a51254cd2cf3c1ea42a3381/CPAC/utils/utils.py#L172-L176 https://github.com/FCP-INDI/C-PAC/blob/35d669bd472006898a51254cd2cf3c1ea42a3381/CPAC/utils/outputs.py#L77-L79 to auto-rename the pre-this-PR motion files. I think we can just delete those parts of the code now that this is built.

File: CPAC/func_preproc/func_preproc.py
Comment (sgiavasis) [reply CPAC/func_preproc/func_preproc.py:1385]: We should keep this comment in, though I cannot seem to trace what the original concern was. Maybe about directly connecting this on its own just to test it? I need to find ou t.

Comment (shnizzedy) [reply CPAC/func_preproc/func_preproc.py:1385]: It's not just the comment, it's the special case handling 
https://github.com/FCP-INDI/C-PAC/blob/e015640fefcc5f1b7e19a856585aaca88e3a542a/CPAC/func_preproc/func_preproc.py#L1315-L1322 https://github.com/FCP-INDI/C-PAC/blob/e015640fefcc5f1b7e19a856585aaca88e3a542a/CPAC/func_preproc/func_preproc.py#L1343-L1347 https://github.com/FCP-INDI/C-PAC/blob/e015640fefcc5f1b7e19a856585aaca88e3a542a/CPAC/generate_motion_statistics/generate_motion_statistics.py#L178-L186 that's [removed](https://github.com/FCP-INDI/C-PAC/commit/de7134775f9c4c245627abe245b4a5e84ec2b304) ↓ https://github.com/FCP-INDI/C-PAC/blob/92cb0774e7946c9364c0e0556008008c83eff7e0/CPAC/func_preproc/func_preproc.py#L1362-L1368 https://github.com/FCP-INDI/C-PAC/blob/92cb0774e7946c9364c0e0556008008c83eff7e0/CPAC/generate_motion_statistics/generate_motion_statistics.py#L171-L178 is why I ask.

To be fair, I don't see where `subject_id` was used in this special case before this PR, so it might be some legacy pre-1.8 thing

**Comment by tergeorge** [comment]
Tested this PR with both `3dvolreg` and `mcflirt `. In both cases CPAC runs to completion and generates `{sub_ID}_ses-1_task-rest_run-1_desc-summary_motion.tsv` and `{sub_ID}_ses-1_task-rest_run-1_motion.tsv`.

**Comment by shnizzedy** [comment]
> It would be good to condense all of this data into 3 files. There are three kinds:
> 
> One row of numbers with a header denoting each one. We can merge these into one file and change it to a .tsv:
> ```
> sub-*_ses-*_task-rest_run-*_motion-params.txt
> sub-*_ses-*_task-rest_run-*_power-params.txt
> ```
> A 1D file with a single vector, with a row/value per time point from the time series. Again, merge these files, but also create a new header describing each vector, and make it a `.tsv` file:
> ```
> sub-*_ses-*_task-rest_run-*_dvars.1D
> sub-*_ses-*_task-rest_run-*_framewise-displacement-jenkinson.1D
> sub-*_ses-*_task-rest_run-*_framewise-displacement-power.1D
> sub-*_ses-*_task-rest_run-*_max-displacement.1D
> sub-*_ses-*_task-rest_run-*_rels-displacement.1D
> ```
> The `movement-parameters.1D` output, which is often a 6xN matrix, with N = timepoints. This file is used widely downstream as its own output, in the format it is written in, so this can remain the same, although we can also convert it to a `.tsv` file:
> ```
> sub-*_ses-*_task-rest_run-*_movement-parameters.1D
> ```

― https://github.com/FCP-INDI/C-PAC/issues/1576#issue-1005662912

> changes made to func_preproc.py and generate_motion_statistics.py. All .1D files mentioned below combined into 1 .tsv file:
> sub-\*_ses-\*_task-rest_run-\*_framewise-displacement-jenkinson.1D
> sub-\*_ses-\*_task-rest_run-\*_framewise-displacement-power.1D
> sub-\*_ses-\*_task-rest_run-\*_max-displacement.1D
> sub-\*_ses-\*_task-rest_run-\*_rels-displacement.1D
> [~~sub-\*_ses-\*_task-rest_run-\*_movement-parameters.1D~~]
> 
> All 2D matrix values mentioned below combined into 2nd .tsv file:
> sub-\*_ses-\*_task-rest_run-\*_motion-params.txt
> sub-\*_ses-\*_task-rest_run-\*_power-params.txt
> 
> The new files generated are [`motion.tsv`] and [`desc-summary_motion.tsv`].

― https://github.com/FCP-INDI/C-PAC/pull/1624#issue-1061359255

> Flips the priority in the logic from `filtered-movement-parameters` being the special case to `unfiltered-movement-parameters` being the special case (with `movement-parameters` being the other, default case both before and after)

― https://github.com/FCP-INDI/C-PAC/pull/1989#issue-1875413768

Currently testing that outputs are
* `motion.tsv`
* `desc-summary_motion.tsv`
* `movement-parameters.tsv`
and potentially
* `unfiltered-movement-parameters.tsv`
(will update the latter to become `desc-movementParameters_motion.tsv` and `desc-unfilteredMovementParameters_motion.tsv`)

Note: we'll probably want to rename these again soon since [BIDS now uses `motion` as the suffix for motion capture data](https://bids-specification.readthedocs.io/en/latest/modality-specific-files/motion.html#motion-recording-data)

**Comment by sgiavasis** [comment]
Thank you!

> Note: we'll probably want to rename these again soon since [BIDS now uses `motion` as the suffix for motion capture data](https://bids-specification.readthedocs.io/en/latest/modality-specific-files/motion.html#motion-recording-data)

Do we want to just do this now? Is there a defined proper name to use in the spec other than `motion`?

**Comment by shnizzedy** [comment]
I asked in https://github.com/bids-standard/bids-specification/pull/519#issuecomment-1762364858

**Comment by shnizzedy** [comment]
7b33c49 changes the format of `desc-summary_motion.tsv` from

> | Parameters                     | Values               |
> |--------------------------------|----------------------|
> | Mean_Relative_RMS_Displacement | 0.14145283984038426  |
> | Max_Relative_RMS_Displacement  | 1.389282568761795    |
> | Movements_gt_threshold         | 212.0                |
> | Mean_Relative_Mean_Rotation    | 0.017968538324420677 |
> | Mean_Relative_Maxdisp          | 0.19005614973262028  |
> | Max_Relative_Maxdisp           | 3.832                |
> | Max_Abs_Maxdisp                | 4.367                |
> | Max Relative_Roll              | 0.06999999999999999  |
> | Max_Relative_Pitch             | 1.5071999999999999   |
> | Max_Relative_Yaw               | 0.3244               |
> | Max_Relative_dS-I              | 1.4773               |
> | Max_Relative_dL-R              | 0.05430000000000001  |
> | Max_Relative_dP-A              | 0.2651               |
> | Mean_Relative_Roll             | 0.008331283422459894 |
> | Mean_Relative_Pitch            | 0.04126631016042781  |
> | Mean_Relative_Yaw              | 0.026191978609625667 |
> | Mean_Relative_dS-I             | 0.21911791443850265  |
> | Mean_Relative_dL-R             | 0.009462299465240643 |
> | Mean_Relative_dP-A             | 0.10449518716577538  |
> | Max_Abs_Roll                   | 0.0924               |
> | Max_Abs_Pitch                  | 1.7731               |
> | Max_Abs_Yaw                    | 0.3401               |
> | Max_Abs_dS-I                   | 1.581                |
> | Max_Abs_dL-R                   | 0.102                |
> | Max_Abs_dP-A                   | 0.2047               |
> | Mean_Abs_Roll                  | 0.04281546666666667  |
> | Mean_Abs_Pitch                 | 0.09135386666666667  |
> | Mean_Abs_Yaw                   | 0.0942888            |
> | Mean_Abs_dS-I                  | 0.18470373333333334  |
> | Mean_Abs_dL-R                  | 0.0356336            |
> | Mean_Abs_dP-A                  | 0.06858533333333333  |
> | MeanFD_Power                   | 0.3981497083287398   |
> | MeanFD_Jenkinson               | 0.26231529997333336  |
> | rootMeanSquareFD               | 0.5121672578107013   |
> | FDquartile(top1/4thFD)         | 0.46138294591397855  |
> | MeanDVARS                      | 84.98434029333335    |

(36 rows + a header × 2 columns)

to 

> | Mean_Relative_RMS_Displacement | Max_Relative_RMS_Displacement | Movements_gt_threshold | Mean_Relative_Mean_Rotation | Mean_Relative_Maxdisp | Max_Relative_Maxdisp | Max_Abs_Maxdisp | Max Relative_Roll   | Max_Relative_Pitch | Max_Relative_Yaw | Max_Relative_dS-I | Max_Relative_dL-R   | Max_Relative_dP-A | Mean_Relative_Roll   | Mean_Relative_Pitch | Mean_Relative_Yaw    | Mean_Relative_dS-I  | Mean_Relative_dL-R   | Mean_Relative_dP-A  | Max_Abs_Roll | Max_Abs_Pitch | Max_Abs_Yaw | Max_Abs_dS-I | Max_Abs_dL-R | Max_Abs_dP-A | Mean_Abs_Roll       | Mean_Abs_Pitch      | Mean_Abs_Yaw | Mean_Abs_dS-I       | Mean_Abs_dL-R | Mean_Abs_dP-A       | MeanFD_Power       | MeanFD_Jenkinson    | rootMeanSquareFD   | FDquartile(top1/4thFD) | MeanDVARS         |
> |--------------------------------|-------------------------------|------------------------|-----------------------------|-----------------------|----------------------|-----------------|---------------------|--------------------|------------------|-------------------|---------------------|-------------------|----------------------|---------------------|----------------------|---------------------|----------------------|---------------------|--------------|---------------|-------------|--------------|--------------|--------------|---------------------|---------------------|--------------|---------------------|---------------|---------------------|--------------------|---------------------|--------------------|------------------------|-------------------|
> | 0.14145283984038426            | 1.389282568761795             | 212.0                  | 0.017968538324420677        | 0.19005614973262028   | 3.832                | 4.367           | 0.06999999999999999 | 1.5071999999999999 | 0.3244           | 1.4773            | 0.05430000000000001 | 0.2651            | 0.008331283422459894 | 0.04126631016042781 | 0.026191978609625667 | 0.21911791443850265 | 0.009462299465240643 | 0.10449518716577538 | 0.0924       | 1.7731        | 0.3401      | 1.581        | 0.102        | 0.2047       | 0.04281546666666667 | 0.09135386666666667 | 0.0942888    | 0.18470373333333334 | 0.0356336     | 0.06858533333333333 | 0.3981497083287398 | 0.26231529997333336 | 0.5121672578107013 | 0.46138294591397855    | 84.98434029333335 |

(1 row + a header × 36 columns)

**Comment by sgiavasis** [comment]
Amazing! Very happy to merge this in after almost 2 years 😆

**Pull request merged**
